src/ship-gen/shipGenerator.js

```diff
@@ -1,8 +1,8 @@
 import Srand from 'seeded-rand';
 import GameMap from "../gameMap.js";
-import { create2dArray } from '../../utils.js';
 import EntityFactories from '../entityFactories.js';
 import Tiles from './tilefactories';
+import {RoomConstants, BreachRoom, Bridge, RoomTypeFactories, RectangularRoom } from './roomTypes';
 
 export class GeneratorOptions {
```

Comment:
A few of these aren't used but the idea is here. Enemy information may be split out to a populator pass and be more determined by room type.